charts/consul/templates/mesh-gateway-deployment.yaml

```diff
@@ -184,6 +184,12 @@ spec:
       containers:
       - name: mesh-gateway
         image: {{ .Values.global.imageConsulDataplane | quote }}
+        securityContext:
+          capabilities:
+            drop:
+              - ALL
```

Comment:
Noting that we talked a few minutes ago about gating this `drop: [ALL]` behind `not .Values.meshGateway.hostNetwork` so that we still allow that to work for the few users who require it.

This will leave us with more work to do to lock down the security context specifically for that `hostNetwork` use case but will unblock all other use cases who can't upgrade right now due to the fact that mesh gateway doesn't request `NET_BIND_SERVICE` when it's actually required.

Reply:
Sounds great (in this case we would want a unit test to test the conditional. That makes a lot of sense to me, not sure why that wasn't done in the original PR.

Reply:
Updated with fixes from these comments/discussions